lib/evmone/instructions.hpp

```diff
@@ -968,6 +971,30 @@ inline StopToken return_impl(StackTop stack, ExecutionState& state) noexcept
 inline constexpr auto return_ = return_impl<EVMC_SUCCESS>;
 inline constexpr auto revert = return_impl<EVMC_REVERT>;
 
+inline StopToken returncontract(StackTop stack, ExecutionState& state, code_iterator pos) noexcept
+{
+    const auto& offset = stack[0];
+    const auto& size = stack[1];
+
+    if (!check_memory(state, offset, size))
+        return {EVMC_OUT_OF_GAS};
+
+    const auto deploy_container_index = size_t{pos[1]};
+
+    const auto header = read_valid_eof1_header(state.original_code);
+    bytes deploy_container{&state.original_code[header.container_begin(deploy_container_index)],
+        header.container_sizes[deploy_container_index]};
```

Comment:
Shouldn't we have a `container_size()` helper?

Reply:
Added the helper, but we also need to add `code_size()` to `eof` branch.